docs/index.rst

```diff
@@ -78,10 +77,16 @@ Here is a simple Flask-Menu usage example:
 
     from flask import Flask
     from flask import render_template_string
-    from flask_menu import Menu, register_menu
+    from flask_menu import FlaskMenu
 
     app = Flask(__name__)
-    Menu(app=app)
+    FlaskMenu(app=app)
+
+    def init(app):
+        menu = app.extensions["menu"]
+        menu.submenu(".").register("index", "Home")
+        menu.submenu(".first").register("first", "First")
+        menu.submenu(".second").register("second", "Second", order=1)
```

Comment:
```suggestion
    menu = FlaskMenu()

    top_menu = menu.submenu(".")
    top_menu.register("index", "Home")

    # or via decorator
    first_menu = menu.submenu(".first")
    
    @app.route("/first"
    @first_menu.register("First")
    def first_index():
        ...
    
    # Deferred extension initialization
    menu.init_app(app)
```

minor: shouldn't we aim for this kind of usage for the module? Usually with Flask extensions that are used like decorators (e.g. Flask-Limiter and Flask-Talisman), the extension object is instantiated without passing the Flask app, so that it can be imported all over the codebase and be used to decorate blueprint views.

The idea is that all the decorator usage is "lazy", i.e. the extension just stores references to view functions or endpoint names and doesn't depend on application/request context. All the functionality that depends on application context is also "lazy" (e.g. the `Menu.url` property, which calls `url_for`).

Reply:
i don't see an easy solution yet how we can use the Menu() as a decorator, because of the gone `@before_first_request` decorator. this would mean that the menu has to be instantiated on every request. I don't see the benefit for an decorator. 

Should we do that because of the convention that Flask extensions are used that way? Is there a real technical benefit? Is it better, in terms of simpler for the developers who will use the decorator?

Reply:
I will try it. My first thought was that it should not be possible, because otherwise the `before_first_request` approach was useless, but on the other hand it could be possible, but i see problems when it comes to the sorting and nesting of the menus.